dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/TrackerPreheat.java

```diff
@@ -267,6 +267,15 @@ public MetadataIdentifier getCategoryOptionComboIdentifier( CategoryCombo catego
     @Getter
     private final Map<String, Relationship> relationships = new HashMap<>();
 
+    /**
+     * Internal map of all relationships in the DB that are duplications of
+     * relationships present in the payload. Two relationship are duplicated if
+     * they have the same relationshipType, and the same relationshipItems, from
+     * and to.
+     */
+    @Getter
```

Comment:
Do you need the getter? I am asking because I removed getters in the preheat, where I could so we do not expose our mutable collections. I then looked for what client code needed and added these methods instead.

Ideally we would have a way of enforcing that only in the preheat phase can you add things to the preheat (maybe via an interface). All other phases should only get access to immutable data/methods returning immutable data.

Reply:
Removed